code/modules/power/supermatter/supermatter.dm

```diff
@@ -896,6 +896,11 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 /obj/machinery/proc/supermatter_zap(atom/zapstart = src, range = 5, zap_str = 3.2 MEGA JOULES, zap_flags = ZAP_SUPERMATTER_FLAGS, list/targets_hit = list(), zap_cutoff = 1.2 MEGA JOULES, power_level = 0, zap_icon = DEFAULT_ZAP_ICON_STATE, color = null)
 	if(QDELETED(zapstart))
 		return
+	if(!zap_str || zap_str < 0) // Just in case something scales zap_str and zap_cutoff to 0.
+		return
+	if(!zap_cutoff || zap_cutoff < 0)
+		stack_trace("/obj/machinery/supermatter_zap() was called with a non-positive value. This causes a recursion that will hit the recursion limit.")
+		return
```

Comment:
Order, also it couldn't because of your guard logic no sense talking about what used to be the case
Also no sense worrying about nulls here
```suggestion
	if(zap_cutoff <= 0)
		stack_trace("/obj/machinery/supermatter_zap() was called with a non-positive value")
		return
	if(zap_str <= 0) // Just in case something scales zap_str and zap_cutoff to 0.
		return
```